test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,208 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	farController "github.com/medik8s/fence-agents-remediation/controllers"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+	farUtils "github.com/medik8s/fence-agents-remediation/test/e2e/utils"
+)
+
+const (
+	// testNamespace  = "fence-agents-remediation"
+	testNamespace  = "openshift-operators"
+	fenceAgentIPMI = "fence_ipmilan"
+
+	// eventually parameters
+	timeout      = 2 * time.Minute
+	pollInterval = 10 * time.Second
+	offsetExpect = 1
+)
+
+var _ = Describe("FAR E2e", func() {
+	testShareParam := map[v1alpha1.ParameterName]string{
+		"--username": "admin",
+		"--password": "password",
+		"--action":   "reboot",
+		"--ip":       "192.168.111.1",
+		"--lanplus":  "",
+	}
+	testNodeParam := map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string{
+		"--ipport": {
+			"master-0": "6230",
+			"master-1": "6231",
+			"master-2": "6232",
+			"worker-0": "6233",
+			"worker-1": "6234",
+			"worker-2": "6235",
+		},
+	} // get ports
+	Context("fence agent - fence_ipmilan", func() {
+		var (
+			nodeBootTimeBefore metav1.Time
+			far                *v1alpha1.FenceAgentsRemediation
+			cond               corev1.NodeCondition
+			errBoot            error
+		)
+		testNode := &corev1.Node{}
+		nodes := &corev1.NodeList{}
+		BeforeEach(func() {
+			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
+			if len(nodes.Items) <= 1 {
+				Skip("there is one or less available nodes in the cluster")
+			}
+			//TODO: Randomize the node selection
+			// Use FA on the first node - master-0
+			testNode = &nodes.Items[0]
+			log.Info("Testing Node", "Node name", testNode.Name)
+
+			// save the node's boot time prior to the fence agent call
+			if cond, errBoot = getKubeletReadyCondition(testNode.Name); errBoot != nil {
```

Comment:
I don't think that getting the last transition of the Ready condition is necessarily equivalent to the node's boot time.
[Here](https://github.com/medik8s/self-node-remediation/blob/main/e2e/self_node_remediation_test.go#L412) is how we get the boot time in SNR

Reply:
> I don't think that getting the last transition of the Ready condition is necessarily equivalent to the node's boot time.

What makes you think this way?

Reply:
I agree but the comment around `getKubeletReadyCondition`.
If the reboot is faster than waiting around 40s until an unresponsive node is marked as not ready, then why this test will fail? IIUC my test will wait more time which is safer than finish testing too early.
It might means that my test will unnecessary last longer. ATM I don't see this as a problem, but when the E2E tests will be more complicated that testing this might be time consuming :thinking

Reply:
I expected that you test that the transition time to ready status is after start test time. When the node never gets unready that will fail.

Reply:
> When the node never gets unready that will fail.

Is that bad? I think it works as I would expect. 
Is there a scenario where a node has been remediated and in the process of remediation the node never gets to _Not Ready_ status?

Reply:
> Is that bad? I think it works as I would expect.

You want to know if FAR works, right? You want to do that by checking that the Node rebooted, right? You want to do that by looking at the Ready condition, right? So when that condition never changes, how do you that the node rebooted and far did something?

> Is there a scenario where a node has been remediated and in the process of remediation the node never gets to Not Ready status?

When remediation is faster than 40s: yes

Reply:
> When remediation is faster than 40s: yes

Now, I got it. Thanks 👍🏻

Reply:
I am reverting to verify reboot by looking for boot time rather than using the getKubeletReadyCondition